api_tests/nodes/views/test_node_sparse_fieldsets.py

```diff
@@ -115,61 +132,95 @@ def test_embed_sparse_same_type(self):
 
-    def test_multiple_sparse_types(self):
-        url = '{}?fields[nodes]=contributors,title&embed=contributors&fields[contributors]=bibliographic'.format(self.url)
-        res = self.app.get(url)
+    #   test_multiple_sparse_types
+        res_url = '{}?fields[nodes]=contributors,title&embed=contributors&fields[contributors]=bibliographic'.format(url)
+        res = app.get(res_url)
         node_json = res.json['data']
 
         assert set(node_json.keys()) == set(['links', 'type', 'id', 'attributes', 'embeds', 'relationships'])
         assert node_json['attributes'].keys() == ['title']
         assert len(node_json['embeds']['contributors']['data']) == 1
-        assert node_json['embeds']['contributors']['data'][0]['id'] == '{}-{}'.format(self.node._id, self.user._id)
+        assert node_json['embeds']['contributors']['data'][0]['id'] == '{}-{}'.format(node._id, user._id)
         assert node_json['embeds']['contributors']['data'][0]['attributes'].keys() == ['bibliographic']
 
-    def test_update_with_sparse_fields(self):
-        url = '{}?fields[nodes]='.format(self.url)
-        old_title = self.node.title
+    def test_update_with_sparse_fields(self, app, user, node, url):
+        res_url = '{}?fields[nodes]='.format(url)
+        old_title = node.title
         payload = {'data': {
-            'id': self.node._id,
+            'id': node._id,
             'type': 'nodes',
             'attributes': {
                 'title': 'new title'
             }
         }}
-        res = self.app.patch_json_api(url, payload, auth=self.user.auth)
+        res = app.patch_json_api(res_url, payload, auth=user.auth)
         assert res.status_code == 200
         assert res.json['data']['attributes'] == {}
-        self.node.reload()
-        assert self.node.title != old_title
-
-
-class TestSparseViewOnlyLinks(ViewOnlyTestCase):
-
-    def test_sparse_fields_with_anonymous_link(self):
-        res = self.app.get(self.private_node_one_url, {
-            'view_only': self.private_node_one_anonymous_link.key,
+        node.reload()
+        assert node.title != old_title
+
+@pytest.mark.django_db
+class TestSparseViewOnlyLinks:
+
+    @pytest.fixture()
+    def creation_user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def viewing_user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def contributing_read_user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def contributing_write_user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def private_node_one(self, creation_user, contributing_read_user, contributing_write_user):
+        private_node_one = ProjectFactory(is_public=False, creator=creation_user, title='Private One')
+        private_node_one.add_contributor(contributing_read_user, permissions=[permissions.READ], save=True)
+        private_node_one.add_contributor(contributing_write_user, permissions=[permissions.WRITE], save=True)
```

Comment:
This gives only write permission, which does not implicitly give read. While it probably won't cause a problem since model level things will get it implicitly, it is best to add READ anywhere you explicitly add WRITE.